tools/docker/bin/docker-db-autoupgrade.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Automatically upgrade the MySQL/MariaDB databases when starting the container
+# This is required on every major database version bump.
+# The script will wait until we can connect to the database server in the main container,
+# run the upgrade command and then exit.
+
+CREDENTIALS_PARAMS="-uroot -p${MYSQL_ROOT_PASSWORD} -hdb"
+
+# Wait until server is up
+while ! mysql $CREDENTIALS_PARAMS ${MYSQL_DATABASE} -e "SELECT 1 FROM wp_posts LIMIT 1" > /dev/null 2>&1
```

Comment:
This loops forever if WordPress hasn't been installed.

You don't really even need to do a query, all we're doing is testing the connection so the command can just be "quit".
```suggestion
while ! mysql $CREDENTIALS_PARAMS ${MYSQL_DATABASE} -e 'quit' > /dev/null 2>&1
```